keypoints/include/pcl/keypoints/impl/trajkovic_3d.hpp

```diff
@@ -230,8 +246,16 @@ pcl::TrajkovicKeypoint3D<PointInT, PointOutT, NormalT>::detectKeypoints (PointCl
   const int width (input_->width);
   const int height (input_->height);
 
-#ifdef _OPENMP
-#pragma omp parallel for shared (output) num_threads (threads_)
+#if defined _OPENMP && _OPENMP <= 201307 || (__GNUC__ >= 6 && __GNUC__ < 9)
```

Comment:
It will help to create a common macro with the rationale behind the values (ie. the link regarding GCC's internal impl) and use it wherever `const` qualified variables are `shared`

Reply:
Any good idea for a macro? As we can't use a macro within a macro...

Reply:
I meant something along the lines of
```
#if PCL_OPENMP_SHARED_CONST_VARIABLES
#pragma form1
#else
#pragma form2
#endif
```

My naming scheme feels overly verbose to me, so feel free to pick another name. This is just so the macro doesn't become opaque to future readers

Reply:
Imho it is not worth to create a extra header file just for this shortcut.